src/System.Collections/tests/Generic/Queue/QueueTests.cs

```diff
@@ -31,6 +32,95 @@ public void Ctors()
         }
 
         [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+
+            Assert.Equal(collection, q);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_EnqueueDequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            q.Enqueue(int.MaxValue);
+            q.Enqueue(int.MinValue);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+            Assert.Equal(int.MaxValue, q.Dequeue());
+            Assert.Equal(int.MinValue, q.Dequeue());
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_Dequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            IEnumerator<int> e = collection.GetEnumerator();
+
+            e.MoveNext();
+            q.Dequeue();
+
+            while (e.MoveNext())
+            {
+                Assert.Equal(e.Current, q.Dequeue());
+            }
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_DequeueEnqueue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            IEnumerator<int> e = collection.GetEnumerator();
+
+            e.MoveNext();
+            q.Dequeue();
+
+            q.Enqueue(int.MaxValue);
+
+            while (e.MoveNext())
+            {
+                Assert.Equal(e.Current, q.Dequeue());
+            }
+            Assert.Equal(int.MaxValue, q.Dequeue());
+        }
+
+        public static IEnumerable<object[]> Collections
+        {
+            get
+            {
+                foreach (int size in new int[] { 1, 2, 3, 4, 5 })
+                {
+                    yield return new object[] { Enumerable.Range(0, size).ToArray() };
+                    yield return new object[] { Enumerable.Range(0, size).ToList() };
+                    yield return new object[] { new Collection<int>(Enumerable.Range(0, size).ToList()) };
+                    yield return new object[] { new ReadOnlyCollection<int>(Enumerable.Range(0, size).ToList()) };
+                    yield return new object[] { Enumerable.Range(0, size) };
+                    yield return new object[] { CreateIteratorCollection(size) };
```

Comment:
Enumerable.Range and CreateIteratorCollection are currently pretty much the same. Any future change to the former is likely to make it pretty much like one of the others. Maybe skip out Range to compensate for the increased cost if adding the other sizes I suggest.